qiskit_experiments/analysis/curve_analysis.py

```diff
@@ -0,0 +1,796 @@
+                    SeriesDef(
+                        name="x",
+                        param_names=["a", "freq", "phase", "b"],
+                        fit_func_index=0,
+                        filter_kwargs={"pulse": "x"}
+                    ),
+                    SeriesDef(
+                        name="y",
+                        param_names=["a", "freq", "phase", "b"],
+                        fit_func_index=1,
+                        filter_kwargs={"pulse": "y"}
+                    )
+                ]
+
+                __fit_funcs__ = [library.cos, library.sin]
+
+                __param_names__ = ["a", "freq", "phase", "b"]
+
+    Notes:
+        This CurveAnalysis class provides several private methods that subclasses can override.
+
+        - Customize figure generation:
+            Override :meth:`~self._create_figures`. For example, here you can create
+            arbitrary number of new figures or upgrade the default figure appearance.
+
+        - Customize pre-data processing:
+            Override :meth:`~self._data_pre_processing`. For example, here you can
+            take a mean over y values for the same x value, or apply smoothing to y values.
+
+        - Customize post-analysis data processing:
+            Override :meth:`~self._post_processing`. For example, here you can
+            calculate new entity from fit values. Such as EPC of RB experiment.
+
+        - Customize fitting options:
+            Override :meth:`~self._setup_fitting`. For example, here you can
+            calculate initial guess from experiment data and setup fitter options.
+
+        - Customize data processor calibration:
+            Override :meth:`~Self._calibrate_data_processor`. This is special subroutine
+            that is only called when a DataProcessor instance is used as the data processor.
+            You can take arbitrary data from experiment result and setup your processor.
+
+        Note that other private methods are not expected to be overridden.
+        If you forcibly override these methods, the behavior of analysis logic is not well tested
+        and we cannot guarantee it works as expected (you may suffer from bugs).
+        Instead, you can open an issue in qiskit-experiment github to upgrade this class
+        with proper unittest framework.
+
+        https://github.com/Qiskit/qiskit-experiments/issues
+    """
+
+    #: str: Metadata key representing a scanned value.
+    __x_key__ = "xval"
+
+    #: List[SeriesDef]: List of mapping representing a data series
+    __series__ = None
+
+    #: List[Callable]: A callback function to define the expected curve
+    __fit_funcs__ = None
+
+    #: List[str]: Parameter name list
+    __param_names__ = list()
+
+    # Callable: Default curve fitter. This can be overwritten.
+    __base_fitter__ = scipy_curve_fit_wrapper
+
+    # Union[Callable, DataProcessor]: Data processor to format experiment data.
+    __default_data_processor__ = level2_probability
```

Comment:
Can you replace this with
```suggestion
    __default_data_processor__ = DataProcessor("counts", [Probability("1")])
```
and add some tests?

Reply:
"1" is qubit number dependent. For two qubit RB, this can be "11". Is it possible to pass the target label as a processor runtime option?

Reply:
Currently, you cannot pass "11" at run-time as this goes against the design `def __call__(self, datum)` we now have. We could extend this so that the data processor has `def __call__(self, datum: Dict[str, Any], **options)` which also means that the nodes need to be able to take options.

Reply:
So we will remove `level2_probability` after the processor is updated? Currently both 1Q and 2Q RB may use the same analysis, thus we cannot hardcode the label here.

Reply:
Perhaps we need to add `data_processor_option` to `SeriesDef`, otherwise we cannot pass any data to processor now. Once we set this option we can encode the target label in metadata. I'll update the logic shortly.

Reply:
Can you check this update? 8de5e37 @eggerdj @gadial

Reply:
I don't think there should be a default here, a subclass should have to specify its data processor.

Reply:
The default is to avoid always having runtime option. i.e. This is kind of `__init__(data_processor = level2_probability)`, but these analysis doesn't have __init__ method since they are attached to experiment class attribute and directly initialized without args by the composite experiment. All subclasses can override the default data processor (depends on expected measurement request).